query-engine/js-connectors/src/proxy.rs

```diff
@@ -349,7 +350,17 @@ impl From<FlavourSpecificResultSet> for QuaintResultSet {
         let names = js_result_set.column_names;
         let rows = js_result_set.rows.iter_mut().map(to_quaint_row).collect();
 
-        QuaintResultSet::new(names, rows)
+        let last_insert_id = js_result_set.last_insert_id.and_then(|id| id.parse::<u64>().ok());
+        let mut quaint_result_set = QuaintResultSet::new(names, rows);
+
+        // Not a fan of this (extracting the `Some` value from an `Option` and pass it to a method that creates a new `Some` value),
+        // but that's Quaint's ResultSet API and that's how the MySQL connector does it.
+        // Sqlite, on the other hand, uses a `last_insert_id.unwrap_or(0)` approach.
+        if let Some(last_insert_id) = last_insert_id {
+            quaint_result_set.set_last_insert_id(last_insert_id);
+        }
+
+        quaint_result_set
```

Comment:
This comment can be removed after the review.
You can take a look at how MySQL and SQLite connectors handle this.
- MySQL: https://github.com/prisma/prisma-engines/blob/1548dc15e415a2298c25af9b04f33bba3d28b998/quaint/src/connector/mysql.rs#L498-L500
- SQLite: https://github.com/prisma/prisma-engines/blob/032ea5987050b4ef0b92272bc00f1e50ff02798a/quaint/src/connector/sqlite.rs#L192


cc @Weakky.